File: variants/java-gradle/aimax/src/main/java/net/pijcke/nim/aimax/MaxAI.java
package net.pijcke.nim.aimax;

import net.pijcke.nim.rules.Nim;
import net.pijcke.nim.rules.Player;

public class MaxAI implements Player {

    @Override
    public String name() {
        return "MaxAI";
    }

    @Override
    public int play(int sticks) {
        return Math.min(Nim.MAX_STICKS_PER_TURN, sticks);
    }

}

File: variants/kotlin-gradle/nim/aimax/src/main/kotlin/net/pijcke/nim/aimax/MaxAi.kt
package net.pijcke.nim.aimax

import net.pijcke.nim.rules.Player
import net.pijcke.nim.rules.Nim
import kotlin.math.min

class MaxAi : Player {

    override fun name() = "Max AI"

    override fun play(sticks: Int) = min(sticks, Nim.MAX_STICKS_PER_TURN)

}